server/stream.go

```diff
@@ -3382,9 +3382,8 @@ func (mset *stream) setStartingSequenceForSources(iNames map[string]struct{}) {
 			delete(iNames, indexName)
 		} else if indexName == _EMPTY_ && streamName != _EMPTY_ {
 			for iName := range iNames {
-				if streamName == mset.sources[iName].name ||
-					(mset.streamSource(iName).External != nil && streamName == mset.sources[iName].name+":"+getHash(mset.streamSource(iName).External.ApiPrefix)) {
-					si := mset.sources[indexName]
+				if si := mset.sources[iName]; si != nil && (streamName == si.name ||
+					(mset.streamSource(iName).External != nil && streamName == si.name+":"+getHash(mset.streamSource(iName).External.ApiPrefix))) {
```

Comment:
Actually this should be si.External != nil

Reply:
Same with last one, I pasted exact code to use.

Reply:
Actually I am wrong here, but we should only call mset.streamSource(iName) once since its a linear walk.